File: 001-020/012.js
//約数の数を数える
var divisor_count = function (num) {
  if (num === 1) {
    return 1;
  }
  if (num < 1) {
    return 0;
  }
  var result = 1;

  var calcNum = num;
  var count = 0;

  while (calcNum % 2 === 0) {
    calcNum /= 2;
    count++;
  }

  count === 0 || (result *= count + 1);

  for (var i = 3; i <= calcNum; i += 2) {
    count = 0;
    while (calcNum % i === 0) {
      calcNum /= i;
      count++;
    }
    count === 0 || (result *= count + 1);
  }

  return result;
};


var count = 500;
var triangle = 1;

for (var i = 2; divisor_count(triangle) <= count; i++) {
  triangle += i;
}

console.log(triangle);

File: 011-020/012.py
# 約数の数を数える
def divisor_count(num):
    if num == 1:
        return 1
    if num < 1:
        return 0
    result = 1

    calcNum = num
    count = 0

    while calcNum % 2 == 0:
        calcNum //= 2
        count += 1

    if count != 0:
        result *= count + 1

    suggest_prime = 3
    while suggest_prime <= calcNum:
        count = 0
        while calcNum % suggest_prime == 0:
            calcNum //= suggest_prime
            count += 1
        if count != 0:
            result *= count + 1
        suggest_prime += 2

    return result

count = 500
triangle = 1

tmp_num = 2
while divisor_count(triangle) <= count:
    triangle += tmp_num
    tmp_num += 1

print(triangle)
